allennlp/pretrained.py

```diff
@@ -0,0 +1,72 @@
+# pylint: disable=line-too-long,invalid-name
+
+from allennlp.predictors import Predictor
+from allennlp.models.archival import load_archive
+
+class PretrainedModel:
+    """
+    A pretrained model is determined by both an archive file
+    (representing the trained model)
+    and a choice of predictor.
+    """
+    def __init__(self, archive_file: str, predictor_name: str) -> None:
+        self.archive_file = archive_file
+        self.predictor_name = predictor_name
+
+    def predictor(self) -> Predictor:
+        archive = load_archive(self.archive_file)
+        return Predictor.from_archive(archive, self.predictor_name)
+
+
+#### Models in the demo ####
+def srl_with_elmo_luheng_2018() -> Predictor:
+    model = PretrainedModel('https://s3-us-west-2.amazonaws.com/allennlp/models/srl-model-2018.05.25.tar.gz',
+                            'semantic-role-labeling')
+    return model.predictor()
+
+def bidirectional_attention_flow_seo_2017() -> Predictor:
```

Comment:
These should be more specific `Predictor` types, so that you can get better code completion on the `predict()` methods.